File: 2021-08-05 (day18)/Hello/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
function twoSum(nums, target) {
  const cache = {};

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] in cache) {
      return [i, cache[nums[i]]];
    }

    cache[target - nums[i]] = i;
  }
}

File: 2021-08-05 (day18)/Hello/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut cache = HashMap::new();

        for (index, num) in nums.iter().enumerate() {
            if let Some(pre_index) = cache.get(num) {
                return vec![*pre_index, index as i32];
            }

            cache.insert(target - num, index as i32);
        }

        unreachable!()
    }
}
